src/Tickets/Emails/Dispatcher.php

```diff
@@ -0,0 +1,507 @@
+
+		$email_slug = $this->get_email()->slug;
+
+		/**
+		 * Filter the attachments for the particular email.
+		 *
+		 * @since TBD
+		 *
+		 * @param array          $attachments The attachments.
+		 * @param Email_Abstract $this        The email object.
+		 */
+		$attachments = apply_filters( "tec_tickets_emails_dispatcher_{$email_slug}_attachments", $attachments, $this );
+
+		return (array) $attachments;
+	}
+
+	/**
+	 * Get email recipient.
+	 *
+	 * @since TBD
+	 *
+	 * @return ?string The email recipient.
+	 */
+	public function get_to(): ?string {
+		/**
+		 * Allow filtering the email recipient globally.
+		 *
+		 * @since TBD
+		 *
+		 * @param string         $to   The email recipient.
+		 * @param Email_Abstract $this The email object.
+		 */
+		$to = apply_filters( 'tec_tickets_emails_dispatcher_to', $this->to, $this );
+
+		$email_slug = $this->get_email()->slug;
+
+		/**
+		 * Allow filtering the email recipient for the particular email.
+		 *
+		 * @since TBD
+		 *
+		 * @param string         $to   The email recipient.
+		 * @param Email_Abstract $this The email object.
+		 */
+		$to = apply_filters( "tec_tickets_emails_dispatcher_{$email_slug}_to", $to, $this );
+
+		return null === $to ? $to : (string) $to;
+	}
+
+	/**
+	 * Sets where we will dispatch to.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $value
+	 */
+	public function set_to( string $value ): void {
+		$this->to = $value;
+	}
+
+	/**
+	 * Get the subject of the email.
+	 *
+	 * @since TBD
+	 *
+	 * @return ?string
+	 */
+	public function get_subject(): ?string {
+		/**
+		 * Allow filtering the email subject globally.
+		 *
+		 * @since TBD
+		 *
+		 * @param string         $subject  The email subject.
+		 * @param Email_Abstract $this     The email object.
+		 */
+		$subject = apply_filters( 'tec_tickets_emails_dispatcher_subject', $this->subject, $this );
+
+		$email_slug = $this->get_email()->slug;
+
+		/**
+		 * Allow filtering the email subject.
+		 *
+		 * @since TBD
+		 *
+		 * @param string         $subject  The email subject.
+		 * @param Email_Abstract $this     The email object.
+		 */
+		$subject = apply_filters( "tec_tickets_emails_dispatcher_{$email_slug}_subject", $subject, $this );
+
+		return null === $subject ? $subject : (string) $subject;
+	}
+
+	/**
+	 * Sets the subject for the dispatcher.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $value
+	 */
+	public function set_subject( string $value ): void {
+		$this->subject = $value;
+	}
+
+	/**
+	 * Get the content of the dispatcher.
+	 *
+	 * @since TBD
+	 *
+	 * @return ?string
+	 */
+	public function get_content(): ?string {
+		/**
+		 * Allow filtering the email content globally.
+		 *
+		 * @since TBD
+		 *
+		 * @param string         $content The email subject.
+		 * @param Email_Abstract $this    The email object.
+		 */
+		$content = apply_filters( 'tec_tickets_emails_dispatcher_content', $this->content, $this );
```

Comment:
So, as far as I understand, we're having this coming from the email abstract class, where we already have two filters for the content (particular and global). What's the point of having this here as well?

I think these things makes it more confusing, adding the `dispatcher` as part of the naming. It should be clear that the email content is populated via the email class, and that this is just a wrapper to send (or dispatch, how you now named it).